pkg/protocol/commitment_verifier.go

```diff
@@ -32,68 +36,38 @@ func NewCommitmentVerifier(mainEngine *engine.Engine, lastCommonCommitmentBefore
 
 	return &CommitmentVerifier{
 		engine:                  mainEngine,
-		cumulativeWeight:        lastCommonCommitmentBeforeFork.CumulativeWeight(),
 		validatorAccountsAtFork: lo.PanicOnErr(mainEngine.Ledger.PastAccounts(accountsAtForkingPoint.IDs(), lastCommonCommitmentBeforeFork.Slot())),
 		// TODO: what happens if the committee rotated after the fork?
 	}, nil
 }
 
-func (c *CommitmentVerifier) verifyCommitment(commitment *model.Commitment, attestations []*iotago.Attestation, merkleProof *merklehasher.Proof[iotago.Identifier]) (blockIDsFromAttestations iotago.BlockIDs, cumulativeWeight uint64, err error) {
-	// 1. Verify that the provided attestations are indeed the ones that were included in the commitment.
-	tree := ads.NewMap[iotago.Identifier](mapdb.NewMapDB(),
-		iotago.Identifier.Bytes,
-		iotago.IdentifierFromBytes,
-		iotago.AccountID.Bytes,
-		iotago.AccountIDFromBytes,
-		(*iotago.Attestation).Bytes,
-		iotago.AttestationFromBytes(c.engine),
-	)
+// verifyCommitment verifies the given commitment and returns the blockIDs and cumulative weight of the attestations.
+func (c *CommitmentVerifier) verifyCommitment(commitment *Commitment, attestations []*iotago.Attestation, merkleProof *merklehasher.Proof[iotago.Identifier]) (blockIDsFromAttestations iotago.BlockIDs, cumulativeWeight uint64, err error) {
```

Comment:
Please add the comments back in this function. They are critical to understand the behavior especially in certain edge cases and serve as documentation within the code. I don't understand why you removed them in first place...

Reply:
I am not sure which comments you mean as this method didn't have a comment before.

If you are talking about the comments about the cumulative weight and the corresponding cutoff condition, then I don't get it either because this calculation and the corresponding condition where replaced by a different check?!?